### PR TITLE
format amount

### LINKED ISSUE #102: AmountInput: format number properly
When entering the sats, show delimiter & remove the zero (make it blank at first)

### PR DESCRIPTION
**Pull request by cstenglein**
Test still failing => Draft

fixes #102

### CODE REVIEW

**Comment by cstenglein** [comment]
TODO:

- [ ] fix btc input (no more than 8 decimal places)
- [ ] remove console.log
- [ ] clean up code

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 1]
```diff
@@ -14,33 +13,75 @@ export type Props = {
 
 const AmountInput: FC<Props> = ({ amount, register, errorMessage }) => {
   const { t } = useTranslation();
+  const [amountInput, setAmountInput] = useState<string>(
+    amount ? "" + amount : ""
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:17]: Hm, looks weird. Is the rsult just`"${amount}"`?

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 2]
```diff
@@ -14,33 +13,75 @@ export type Props = {
...
+
+  const { onChange } = register;
 
-  const ButtonToggleUnit: FC = () => {
-    const { unit, toggleUnit } = useContext(AppContext);
+  const toggleHandler = () => {
+    let formattedValue = amountInput;
+    if (unit === Unit.BTC) {
+      formattedValue = new Intl.NumberFormat("en-US").format(
+        convertBtcToSat(+formattedValue)!
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:27]: Wo we need the `!`?

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 3]
```diff
@@ -14,33 +13,75 @@ export type Props = {
...
+      );
+      setAmountInput(formattedValue);
+    } else {
+      formattedValue = formattedValue.replace(/,|\./g, "");
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:31]: Wanna add a comment what this reges does?

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 4]
```diff
@@ -14,33 +13,75 @@ export type Props = {
...
+    }
+    toggleUnit();
+    onChange({ target: { value: formattedValue } });
+  };
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    let value = e.target.value;
+    value = value.replace(/[^0-9.,]/, "");
+    if (unit === Unit.SAT) {
+      value = value.replace(/,|\./g, "");
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:46]: Or if we use these regexs in different places, can those be defiend once and commented and placed in needed places?

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 5]
```diff
@@ -14,33 +13,75 @@ export type Props = {
...
+      value = new Intl.NumberFormat("en-US").format(+value);
+    } else {
+      // remove commas and replace ".." with "."
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:49]: Ah, here's a comment :D

File: src/components/Shared/AmountInput/AmountInput.tsx (hunk 1 above)
Comment (cstenglein) [reply src/components/Shared/AmountInput/AmountInput.tsx:17]: Would lead to "undefined" in the input if you would use it that way and not specify amount.

Comment (cstenglein) [reply src/components/Shared/AmountInput/AmountInput.tsx:17]: My bad, works then I leave the check ;)

File: src/components/Shared/AmountInput/AmountInput.tsx (hunk 3 above)
Comment (cstenglein) [reply src/components/Shared/AmountInput/AmountInput.tsx:31]: Done :+1:

File: src/components/Shared/AmountInput/AmountInput.tsx (hunk 2 above)
Comment (cstenglein) [reply src/components/Shared/AmountInput/AmountInput.tsx:27]: Yeah, but changed it to `0`

File: src/components/Shared/AmountInput/AmountInput.tsx [hunk 6]
```diff
@@ -1,46 +1,91 @@
-import type { FC } from "react";
```
Comment (escapedcat) [inline src/components/Shared/AmountInput/AmountInput.tsx:1]: Should this have a test or is it covered by the other updated tests?

File: src/util/format.ts [hunk 7]
```diff
@@ -28,3 +28,29 @@ export function convertToString(unit: Unit, num: number | null): string {
   }
   return num.toFixed(8).toString();
 }
+
+export function formatAmount(value: string, unit: Unit): string {
```
Comment (escapedcat) [inline src/util/format.ts:32]: Should this have a test or is it covered by the other updated tests?

**Review by escapedcat — APPROVED** [review]

File: src/util/format.ts (hunk 7 above)
Comment (cstenglein) [reply src/util/format.ts:32]: Will be done in #306 :+1:

File: src/components/Shared/AmountInput/AmountInput.tsx (hunk 6 above)
Comment (cstenglein) [reply src/components/Shared/AmountInput/AmountInput.tsx:1]: Will be done in #306 :+1:

**Pull request merged**